api/envoy/type/matcher/v3/http_inputs.proto

```diff
@@ -55,3 +55,12 @@ message HttpResponseTrailerMatchInput {
   string header_name = 1
       [(validate.rules).string = {well_known_regex: HTTP_HEADER_NAME strict: false}];
 }
+
+// Match input indicates that matching should be done on a specific cookie key.
+// The resulting input string will be all values for the given key joined by a
+// comma, e.g. if the request contains two 'foo' cookie keys with value 'bar'
+// and 'baz', the input string will be 'bar,baz'.
+message HttpRequestCookieMatchInput {
```

Comment:
Can you clarify in the comment the order in which the values are concatenated, please? It maybe worthwhile to clarify that if the backend treats `Cookie: mykey=a;mykey=b` and `Cookie: mykey=b;mykey=a` the same then the matcher may need to include both cases.

Reply:
FYI I've implemented this behavior here, to make it easier to understand : https://github.com/envoyproxy/envoy/pull/16344/commits/2891366d047f8121699448dcc4d45018d631b034